tests/deployment_utils.py

```diff
@@ -128,7 +128,7 @@ def exit(self):
 
 
 @pytest.fixture()
-def execution_environment(request) -> ExecutionEnvironment:
+def execution_environment(request) -> Generator[ExecutionEnvironment, None, None]:
```

Comment:
1. This is unrelated change. In general I am ok to mix annotation in different modules inside one commit, but configs change to class is too big change on its own.
2. pytest fixture are quite special beasts. I don't think we could benefit from this annotation anyway. Does mypy know that the function here and the parameter in test function are connected?

Reply:
mypy flagged these 2 lines as errors, when I checked the test files. I am fine with moving this change to separate PR.